v2/internal/controllers/secrets_from_azure_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1alpha1api20200601"
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1alpha1api20210401"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/conditions"
+)
+
+func Test_WhenObjectDeleted_SecretIsDeleted(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Create a storage account
+	storageKeysSecret := "storagekeys"
+	acct := makeSimpleStorageAccountWithOperatorSpecSecrets(tc, rg, storageKeysSecret, "key1")
+
+	tc.CreateResourceAndWait(acct)
+
+	// Expect that the secret exists and has an ownerReference set. We can't actually ensure
+	// that the resource is deleted in envtest because the controller that performs ownerReference
+	// resource deletion doesn't run in envtest.
```

Comment:
Is this "doesn't run" == "doesn't work" or "doesn't run" == "isn't run (but we could)" ?

Reply:
doesn't work. Will clarify. That component isn't actually part of apiserver and envtest only emulates apiserver, so some of the "background processes" that exist in a normal cluster don't work in envtest